pkg/controller/pxc/controller_test.go

```diff
@@ -57,3 +69,450 @@ var _ = Describe("PerconaXtraDB Cluster", Ordered, func() {
 		Expect(err).To(Succeed())
 	})
 })
+
+var _ = Describe("Finalizer delete-ssl", Ordered, func() {
+	ctx := context.Background()
+
+	const crName = "del-ssl-fnlz"
+	const ns = "del-ssl-fnlz"
+	crNamespacedName := types.NamespacedName{Name: crName, Namespace: ns}
+
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ns,
+			Namespace: ns,
+		},
+	}
+
+	BeforeAll(func() {
+		By("Creating the Namespace to perform the tests")
+		err := k8sClient.Create(ctx, namespace)
+		Expect(err).To(Not(HaveOccurred()))
+
+		_, err = envtest.InstallCRDs(cfg, envtest.CRDInstallOptions{
+			Paths: []string{filepath.Join("testdata", "cert-manager.yaml")},
+		})
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterAll(func() {
+		By("Deleting the Namespace to perform the tests")
+		_ = k8sClient.Delete(ctx, namespace)
+	})
+
+	Context("delete-ssl finalizer specified", Ordered, func() {
+
+		cr, err := readDefaultCR(crName, ns)
+		cr.Finalizers = append(cr.Finalizers, "delete-ssl")
+		cr.Spec.SSLSecretName = "cluster1-ssl"
+		cr.Spec.SSLInternalSecretName = "cluster1-ssl-internal"
```

Comment:
You have to update `cr` after you assert(`"should read default cr.yaml"`) it is Ok, because if it fails this will panic because `cr` will be nill if some error is returned.

Reply:
so you mean I should first and then update CR to catch errors. Yes?
```
It("should read default cr.yaml", func() {
  Expect(err).NotTo(HaveOccurred())
  })
```

Reply:
145307b